File: rust/src/models/subscription_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{pagination_response::PaginationResponse, subscription::Subscription};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SubscriptionListResponse {
    pub data: Vec<Subscription>,

    pub pagination_meta: PaginationResponse,
}

impl SubscriptionListResponse {
    pub fn new(data: Vec<Subscription>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/subscriptionListResponse.ts
// this file is @generated
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";
import { type Subscription, SubscriptionSerializer } from "./subscription";

export interface SubscriptionListResponse {
  data: Subscription[];

  paginationMeta: PaginationResponse;
}

export const SubscriptionListResponseSerializer = {
  _fromJsonObject(object: any): SubscriptionListResponse {
    return {
      data: object["data"].map((item: any) =>
        SubscriptionSerializer._fromJsonObject(item)
      ),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: SubscriptionListResponse): any {
    return {
      data: self.data.map((item: any) => SubscriptionSerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
